pkg/ccl/importccl/read_import_mysql.go

```diff
@@ -143,5 +150,197 @@ func mysqlToCockroach(
+			return nil, err
+		}
+		desc.AddColumn(*col)
+		colNames[raw.Name.String()] = col.Name
+	}
+
+	for _, raw := range in.Indexes {
+		var elems tree.IndexElemList
+		for _, col := range raw.Columns {
+			elems = append(elems, tree.IndexElem{Column: tree.Name(colNames[col.Column.String()])})
+		}
+		idx := sqlbase.IndexDescriptor{Name: raw.Info.Name.String(), Unique: raw.Info.Unique}
+		if err := idx.FillColumns(elems); err != nil {
+			return nil, err
+		}
+		if err := desc.AddIndex(idx, raw.Info.Primary); err != nil {
+			return nil, err
+		}
+	}
+
+	// AllocateIDs mutates its receiver. `return desc, desc.AllocateIDs()`
+	// happens to work in gc, but does not work in gccgo.
+	//
+	// See https://github.com/golang/go/issues/23188.
+	err := desc.AllocateIDs()
+	return &desc, err
+}
+
+func mysqlColToCockroach(name string, col mysql.ColumnType) (*tree.ColumnTableDef, error) {
+	def := &tree.ColumnTableDef{Name: tree.Name(name)}
+
+	var length, scale int
+
+	if col.Length != nil {
+		num, err := strconv.Atoi(string(col.Length.Val))
+		if err != nil {
+			return nil, errors.Wrapf(err, "could not parse length for column %q", name)
+		}
+		length = num
+	}
+
+	if col.Scale != nil {
+		num, err := strconv.Atoi(string(col.Scale.Val))
+		if err != nil {
+			return nil, errors.Wrapf(err, "could not parse scale for column %q", name)
+		}
+		scale = num
+	}
+
+	switch col.SQLType() {
+	case mysqltypes.Char:
+		def.Type = strOpt(coltypes.Char, length)
+	case mysqltypes.VarChar:
+		def.Type = strOpt(coltypes.VarChar, length)
+	case mysqltypes.Text:
+		def.Type = strOpt(coltypes.Text, length)
+	case mysqltypes.Blob:
+		def.Type = coltypes.Bytes
+	case mysqltypes.VarBinary:
+		def.Type = coltypes.Bytes
+	case mysqltypes.Binary:
+		def.Type = coltypes.Bytes
+
+	case mysqltypes.Int8:
+		def.Type = coltypes.SmallInt
+	case mysqltypes.Uint8:
+		def.Type = coltypes.SmallInt
+	case mysqltypes.Int16:
+		def.Type = coltypes.SmallInt
+	case mysqltypes.Uint16:
+		def.Type = coltypes.SmallInt
+	case mysqltypes.Int24:
+		def.Type = coltypes.Int
+	case mysqltypes.Uint24:
+		def.Type = coltypes.Int
+	case mysqltypes.Int32:
+		def.Type = coltypes.Int
+	case mysqltypes.Uint32:
+		def.Type = coltypes.Int
+	case mysqltypes.Int64:
+		def.Type = coltypes.BigInt
+	case mysqltypes.Uint64:
+		def.Type = coltypes.BigInt
+
+	case mysqltypes.Float32:
+		def.Type = coltypes.Float4
+	case mysqltypes.Float64:
+		def.Type = coltypes.Double
 
+	case mysqltypes.Decimal:
+		def.Type = decOpt(coltypes.Decimal, length, scale)
+
+	case mysqltypes.Date:
+		def.Type = coltypes.Date
+	case mysqltypes.Time:
+		def.Type = coltypes.Time
+	case mysqltypes.Timestamp:
+		def.Type = coltypes.TimestampWithTZ
+	case mysqltypes.Datetime:
+		def.Type = coltypes.TimestampWithTZ
+	case mysqltypes.Year:
+		def.Type = coltypes.SmallInt
+
+	case mysqltypes.Enum:
+		fallthrough
+	case mysqltypes.Set:
+		fallthrough
+	case mysqltypes.Geometry:
+		fallthrough
+	case mysqltypes.TypeJSON:
+		fallthrough
```

Comment:
we can't do this?

Reply:
Figured punt it to a followup, since I don't know yet if/how our json type matches up.

Reply:
added TODOs for this and `bit`.